### DISCUSSION TITLE
Enumerate enums?

### DISCUSSION THREAD

**Discussion by martinstreicher** [alassek/activerecord-pg_enum#28] [Q&A]
Let's say I define an enum in PG using something like this... 

```
CREATE TYPE public.standard_status_enum AS ENUM (
    '',
    'Active',
    'Active Under Contract',
    'Canceled',
    'Closed',
    'Coming Soon',
    'Delete',
    'Expired',
    'Hold',
    'Incomplete',
    'Pending',
    'Withdrawn'
);
```

... or I use your gem to define the list of legiitimate values. 

Is there a way, akin to `columns_hash` and other inspection methods on `connection` to:
a. Find all the enumerated types defined in the DB? 
b. For each enumerated type, find all the legitimate values defined?

**Comment by martinstreicher** [comment]
It looks like this code (https://github.com/alassek/activerecord-pg_enum/blob/main/lib/active_record/pg_enum/6.1/postgresql_adapter.rb) runs a query to get the enums.

**Reply by martinstreicher** [reply]
I see its used in the monkey patches -- is there a way I can add the `enum_types` method to my model?

**Reply by fazzledev** [reply]
@martinstreicher any luck with this? Is there an api to access the enum values in the database without writing raw SQL?
